dataset/bin.cpp

```diff
@@ -193,11 +193,12 @@ DataArray add_metadata(std::tuple<DataArray, Variable> &&proto,
   bin_sizes = squeeze(bin_sizes, erase);
   const auto end = cumsum(bin_sizes);
   const auto buffer_dim = buffer.dims().inner();
-  // TODO We probably want to omit the coord used for grouping in the non-edge
-  // case, since it just contains the same value duplicated for every row in the
-  // bin.
-  // Note that we should then also recreate that variable in concatenate, to
-  // ensure that those operations are reversible.
```

Comment:
Does `concatenate` restore the coord correctly?

Reply:
No, and in fact I did not understand the point of the original comment any more (and removed it): When you concatenate data arrays (or bins of data arrays) you still have the bin coords, so there is no reversibility issue.

The comment *might* come from the idea that someone could concatenate binned *variables* (rather than data arrays containing binned data), but this seems rare, and the lack of coords is just what you get from working with variables.

Reply:
To be a bit more concrete: The comment may have been written with a reduction operation in mind:
```python
da.bins.concatenate('pixel')  # after this PR we don't know which event is from which pixel
```
But consider the "equivalent" for dense data:
```python
da.sum('pixel')  # don't know which contribution is from which pixel
```

*Iff* there is a user requirement to preserve this a flag could be added.